main.cpp

```diff
@@ -1,63 +1,28 @@
 #include <iostream>
 #include <assert.h>
+#include "ColorPair.h"
 
-namespace TelCoColorCoder
-{
-    enum MajorColor {WHITE, RED, BLACK, YELLOW, VIOLET};
-    enum MinorColor {BLUE, ORANGE, GREEN, BROWN, SLATE};
-
-    const char* MajorColorNames[] = {
-        "White", "Red", "Black", "Yellow", "Violet"
-    };
-    int numberOfMajorColors =
-        sizeof(MajorColorNames) / sizeof(MajorColorNames[0]);
-    const char* MinorColorNames[] = {
-        "Blue", "Orange", "Green", "Brown", "Slate"
-    };
-    int numberOfMinorColors =
-        sizeof(MinorColorNames) / sizeof(MinorColorNames[0]);
-
-    class ColorPair {
-        private:
-            MajorColor majorColor;
-            MinorColor minorColor;
-        public:
-            ColorPair(MajorColor major, MinorColor minor):
-                majorColor(major), minorColor(minor)
-            {}
-            MajorColor getMajor() {
-                return majorColor;
-            }
-            MinorColor getMinor() {
-                return minorColor;
-            }
-            std::string ToString() {
-                std::string colorPairStr = MajorColorNames[majorColor];
-                colorPairStr += " ";
-                colorPairStr += MinorColorNames[minorColor];
-                return colorPairStr;
-            }
-    };
+using namespace TelCoColorCoder;
 
-    ColorPair GetColorFromPairNumber(int pairNumber) {
-        int zeroBasedPairNumber = pairNumber - 1;
-        MajorColor majorColor = 
-            (MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
-        MinorColor minorColor =
-            (MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
-        return ColorPair(majorColor, minorColor);
-    }
-    int GetPairNumberFromColor(MajorColor major, MinorColor minor) {
-        return major * numberOfMinorColors + minor + 1;
-    }
+ColorPair GetColorFromPairNumber(int pairNumber) {
+    int zeroBasedPairNumber = pairNumber - 1;
+    MajorColor majorColor =
+        (MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
+    MinorColor minorColor =
+        (MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
+    return ColorPair(majorColor, minorColor);
 }
+int GetPairNumberFromColor(MajorColor major, MinorColor minor) {
```

Comment:
there is a slight inconsistency inthe APIs. GetColorFromPairNumber() takes in pair number and returns a ColorPair Object, but the reverse operation GetPairNumberFromColor() takes in MajorColor and Minor Color and returns pair number. It is good to have consistent method APIs, for good readability and maintainability.